cpp/include/cugraph/detail/utility_wrappers.hpp

```diff
@@ -65,6 +65,49 @@ void uniform_random_fill(rmm::cuda_stream_view const& stream_view,
 template <typename value_t>
 void scalar_fill(raft::handle_t const& handle, value_t* d_value, size_t size, value_t value);
 
+/**
+ * @brief    Sort a device span
+ *
+ * @tparam      value_t      type of the value to operate on
+ *
+ * @param [in]  handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator,
+ * and handles to various CUDA libraries) to run graph algorithms.
+ * @param[out]  d_span      device span to sort
+ *
+ */
+template <typename value_t>
+void sort(raft::handle_t const& handle, raft::device_span<value_t> d_span);
+
+/**
+ * @brief    Keep unique element from a device span
+ *
+ * @tparam      value_t      type of the value to operate on
+ *
+ * @param [in]  handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator,
+ * and handles to various CUDA libraries) to run graph algorithms.
+ * @param[in]  d_span      device span of unique elements.
+ * @return the number of unique elements.
+ *
+ */
+template <typename value_t>
+size_t unique(raft::handle_t const& handle, raft::device_span<value_t> d_span);
+
+/**
+ * @brief    Increment the values of a device span by a constant value
+ *
+ * @tparam      value_t      type of the value to operate on
+ *
+ * @param [in]  handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator,
+ * and handles to various CUDA libraries) to run graph algorithms.
+ * @param[out]  d_span       device span to update
+ * @param[in]   value        value to be added to each element of the buffer
+ *
+ */
+template <typename value_t>
+void transform_increment(rmm::cuda_stream_view const& stream_view,
+                         raft::device_span<value_t> d_span,
+                         value_t value);
+
```

Comment:
Stream should be the last parameter, I think.  We should do some review of the code and identify other places where we should be passing stream instead of handle.  I think passing the handle into the algorithm is great, since it gives us access to everything.  But I had to do some complex things in MTMG to get some of the lower level functions working in a multi-stream environment because we use the handle too much.  I think we should look at many of the non-public functions and explore passing the comms object and stream instead of passing the handle.

Regarding these wrappers for thrust calls, I think we'll end up with higher quality code if we have function names that are more precise about what we're doing.  I think `sort_ints` might be sufficiently precise... I imagine there are other integer data types that we would want to sort.

Reply:
I went for option 1